Test failure with Astropy 4.1:  test_tabular_fits_2d

With the just-released Astropy 4.1 version, I get the following failure with pytest:

```
_______________________ test_tabular_fits_2d[wavelength] _______________________

tmpdir = local('/tmp/pytest-of-debci/pytest-0/test_tabular_fits_2d_wavelengt0')
spectral_axis = 'wavelength'

    @pytest.mark.parametrize("spectral_axis",
                             ['wavelength', 'frequency', 'energy', 'wavenumber'])
    def test_tabular_fits_2d(tmpdir, spectral_axis):
        wlu = {'wavelength': u.AA, 'frequency': u.GHz, 'energy': u.eV,
               'wavenumber': u.cm**-1}
        # Create a small data set with 2D flux + uncertainty
        disp = np.arange(1, 1.1, 0.01)*wlu[spectral_axis]
        flux = np.ones((3, len(disp)))*np.arange(1, len(disp)+1)**2*1.e-14*u.Jy
        unc = StdDevUncertainty(0.01*np.random.rand(3, len(disp)))
        if spectral_axis not in ('wavelength', ):
            disp = np.flip(disp)
    
        spectrum = Spectrum1D(flux=flux, spectral_axis=disp, uncertainty=unc)
        tmpfile = str(tmpdir.join('_tst.fits'))
        spectrum.write(tmpfile, format='tabular-fits')
    
        # Read it in and check against the original
>       spec = Spectrum1D.read(tmpfile)

/usr/lib/python3/dist-packages/specutils/tests/test_loaders.py:489: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3/dist-packages/astropy/nddata/mixins/ndio.py:58: in __call__
    return registry.read(self._cls, *args, **kwargs)
/usr/lib/python3/dist-packages/astropy/io/registry.py:520: in read
    data = reader(*args, **kwargs)
/usr/lib/python3/dist-packages/specutils/io/default_loaders/tabular_fits.py:103: in tabular_fits_loader
    return generic_spectrum_from_table(tab, wcs=wcs, **kwargs)
/usr/lib/python3/dist-packages/specutils/io/parsing_utils.py:261: in generic_spectrum_from_table
    spectrum = Spectrum1D(flux=flux, spectral_axis=spectral_axis,
/usr/lib/python3/dist-packages/specutils/spectra/spectrum1d.py:200: in __init__
    wcs = gwcs_from_array(self._spectral_axis)
/usr/lib/python3/dist-packages/specutils/utils/wcs_utils.py:210: in gwcs_from_array
    forward_transform = SpectralTabular1D(np.arange(len(array)),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[AttributeError("'SpectralTabular1D' object has no attribute 'points'") raised in repr()] SpectralTabular1D object at 0x7fde1d736640>
points = array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10])
lookup_table = <SpectralAxis [[1.  ],
   [1.01],
   [1.02],
   [1.03],
   [1.04],
   [1.05],
   [1.06],
   [1.07],
   [1.08],
   [1.09],
   [1.1 ]] Angstrom>
method = 'linear', bounds_error = True, fill_value = nan, kwargs = {}
n_models = 1

    def __init__(self, points=None, lookup_table=None, method='linear',
                 bounds_error=True, fill_value=np.nan, **kwargs):
    
        n_models = kwargs.get('n_models', 1)
        if n_models > 1:
            raise NotImplementedError('Only n_models=1 is supported.')
        super().__init__(**kwargs)
        self.outputs = ("y",)
        if lookup_table is None:
            raise ValueError('Must provide a lookup table.')
    
        if not isinstance(lookup_table, u.Quantity):
            lookup_table = np.asarray(lookup_table)
    
        if self.lookup_table.ndim != lookup_table.ndim:
>           raise ValueError("lookup_table should be an array with "
                             "{} dimensions.".format(self.lookup_table.ndim))
E           ValueError: lookup_table should be an array with 1 dimensions.

/usr/lib/python3/dist-packages/astropy/modeling/tabular.py:112: ValueError
```
This happens for all four possible parameters of the test.
Full test log [here](https://ci.debian.net/data/autopkgtest/testing/amd64/s/specutils/7800686/log.gz). Testing with an identical setup, just with Astropy 4.0.1post1 [succeeds](https://ci.debian.net/data/autopkgtest/testing/amd64/s/specutils/7755102/log.gz).

Other environment:
```
platform linux -- Python 3.8.6, pytest-4.6.11, py-1.9.0, pluggy-0.13.0
plugins: remotedata-0.3.2, cov-2.8.1, astropy-header-0.1.2, hypothesis-5.32.1, filter-subpackage-0.1.1, openfiles-0.5.0, asdf-2.7.1, arraydiff-0.3, doctestplus-0.7.0
```

--- a/specutils/io/default_loaders/tabular_fits.py
+++ b/specutils/io/default_loaders/tabular_fits.py
@@ -167,21 +167,10 @@ def tabular_fits_writer(spectrum, file_name, hdu=1, update_header=False, **kwarg
         columns.append(unc.astype(ftype))
         colnames.append("uncertainty")
 
-    # For 2D data transpose from row-major format
-    ndim = 1
-
+    # For > 1D data transpose from row-major format
     for c in range(1, len(columns)):
         if columns[c].ndim > 1:
-            ndim = columns[c].ndim
             columns[c] = columns[c].T
-
-    if ndim > 1:
-        spec_shape = np.ones(ndim, dtype=np.int)
-        spec_shape[0] = -1
-
-        for c in range(1, len(columns)):
-            if columns[c].ndim == 1:
-                columns[c] = columns[c].reshape(spec_shape)
 
     tab = Table(columns, names=colnames, meta=header)
 

